lib/learn_test/strategies/pytest/requirements_checker.rb

```diff
@@ -0,0 +1,75 @@
+require 'open3'
+
+module LearnTest
+  module Pytest
+    class RequirementsChecker
+      def self.check_installation
+        new.check_installation
+      end
+
+      def check_installation
+        PythonChecker.check
+        PipChecker.check
+        PytestChecker.check
+      end
+    end
+
+    class PythonChecker
+      def self.check
+        new.check
+      end
+
+      def check
+        if !python_installed? || !correct_python_version?
+          puts "Please install python 2.7.x or 3.x.x"
+          exit
+        end
+      end
+
+      def python_installed?
+        !`which python`.empty?
+      end
+
+      def correct_python_version?
+        output = Open3.popen2e('python', '--version')
+        version = output[1].read.strip
+        !!version.match(/ 2.7.*| 3.*/)
+      end
+    end
+
+    class PipChecker
+      def self.check
+        new.check
+      end
+
+      def check
+        if !pip_installed?
+          puts "Please ensure pip is installed"
+          exit
+        end
+      end
+
+      def pip_installed?
+        !`which pip`.empty?
+      end
+    end
+
+    class PytestChecker
+      def self.check
+        new.check
+      end
+
+      def check
+        if !pytest_installed?
+          puts "Please ensure pytest is installed"
+          exit
+        end
+      end
+
+      def pytest_installed?
+        !`which pytest`.empty?
+      end
+    end
+
+  end
+end
```

Comment:
See if there's an editor setting you could tweak to ensure files always end with a newline: https://stackoverflow.com/a/729795

Reply:
I had it off for a second for some reason but it has since been turned back on! Great catch.